### PR TITLE
Improve docker setup

### PR DESCRIPTION
**Pull request by luispabon**
This implements a development environment where the site is loaded up simultaneously on php7.0/7.1/7.2.

After bringing up the environment, navigate to:
 * http://localhost:8100 (php 7.0)
 * http://localhost:8101 (php 7.1)
 * http://localhost:8101 (php 7.2)

The environment is built using docker best practices: no more god containers pretending to be a VM. There are now
  * nginx/fpm container pairs
  * A container for building the frontend (node, `gulp watch`, remains open) on startup
  * A container for building the php backend (one shot, does not remain open after it finishes) on startup

PHP containers are using my own http://phpdocker.io base images, this will help reduce size on disk and dramatically speed up environment build.

image

### CODE REVIEW

**Comment by j0k3r** [comment]
f43.me target is still 5.6, so the composer.json shouldn't be updated.
That's why I manually require the new mongo adaptater for Docker: https://github.com/j0k3r/f43.me/blob/master/docker/launch-f43me.sh#L15-L16

**Comment by luispabon** [comment]
Understood, I'm recreating that.

**Comment by luispabon** [comment]
@j0k3r I've updated the PR's description with what I've done, please give it a test and let me know any issues you might find.

**Comment by luispabon** [comment]
I've added php 7.2 also. No xdebug for the time being though.

image

**Comment by j0k3r** [comment]
I did not forget to take a look at your work I just have little time available these past months

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    backend-build:
+      build: docker/php-fpm/71
+      depends_on: [db]
+      container_name: f43-me-backend-build
+      working_dir: /application
+      volumes:
+          - .:/application
+      command: ["docker/backend-build/build.sh"]
+
+    # PHP 7.2 setup - NOTE: no xdebug at this time available for 7.2
```
Comment (j0k3r) [inline docker-compose.yaml]: Xdebug 2.6.0 supports PHP 7.2, could you update your PR?

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      container_name: f43-me-frontend-build
+      working_dir: /application
+      volumes:
+          - .:/application
+      command: ["docker/frontend-build/start.sh"]
+
+    # This container is responsible of initialising the backend code - dies after it's finished doing its thing
+    backend-build:
+      build: docker/php-fpm/71
+      depends_on: [db]
```
Comment (j0k3r) [inline docker-compose.yaml]: Can't this be defined as other array syntax?

```yaml
depends_on:
    - db
```

File: docker/nginx/nginx.conf [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        include fastcgi_params;
+    }
+
+    # Statics
+        location /(bundles|media) {
+        access_log off;
+        expires 30d;
+        try_files $uri @rewriteapp;
+    }
+}
```
Comment (j0k3r) [inline docker/nginx/nginx.conf]: Could you add a new line at the end of that file?

**Pull request closed without merging**